src/sqlfluff/core/linter/runner.py

```diff
@@ -53,16 +57,19 @@ def iter_partials(
         self,
         fnames: List[str],
         fix: bool = False,
-    ) -> Iterator[Tuple[str, Callable]]:
+    ) -> Iterator[Tuple[str, bool, Callable[[], LintedFile]]]:
```

Comment:
Now that this tuple has 4 fields, should we change it to something more readable, like a `NamedTuple`?

Reply:
I'm wondering whether to get rid of the whole `iter_partials` method entirely 🤔. I wonder if there's a significant opportunity for simplification here.

Reply:
Yep, simpler without. I've done a few rounds of simplification. I'm going to come back to it tomorrow and see if I can go further.